### DISCUSSION TITLE
How can I tell tailwind to build the full CSS for local development?

### DISCUSSION THREAD

**Discussion by wilsonparson** [tailwindlabs/tailwindcss#6921] [Help] [answered]
Now that tailwind builds CSS using the `content` key in the config file, I can't find a way to tell it to just build the entire CSS file (i.e., include every class).

I really like having the full CSS at my disposal for local development, because I can very quickly test out classes in browser Dev Tools. I know I can use a CDN, but it would be great if I could pass a flag or a configuration option that tells Tailwind to build everything. Is there a way I can do that?

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your question!
Much appreciated! 🙏

Tailwind 3 has no way to generate a "full" build, it is JIT-only so it will only ever generate the CSS you are actually using.

If you want to ensure it always includes certain classes, you can use the safelisting feature:

https://tailwindcss.com/docs/content-configuration#safelisting-classes

**Comment by adamwathan** [comment]
Check out this discussion for more information:

https://github.com/tailwindlabs/tailwindcss/discussions/6256

**Comment by wilsonparson** [comment]
Thank you, I didn't realize how large the output would be in v3. I'll play with safelisting for local dev.

**Comment by PwrSrg** [comment]
Hi, two questions...

1. Exactly **how many classes** total are there in TW3?
2. Is there a **definitive list** of ALL TW3 classes anywhere online?

Thanks!

**Reply by adamwathan** [reply]
1. I have no idea 😄
2. I guess the docs, but no single list anywhere.

**Comment by cannap** [comment]
maybe this will help a bit https://github.com/rogden/tailwind-config-viewer  not sure it will show everything

//e bit maybe you can check the source and yea get some hints how to get the list :D

**Comment by drexlma** [comment]
very bad, so i can't use this framework in my current project... what a pity.

**Comment by snorkellingcactus** [comment]
Here doing some noise as i suscribe for having the option to generate the full file on dev environment so i can develop super quick using the browser inspector.

**Reply by cannap** [reply]
https://chromewebstore.google.com/detail/atomic-css-devtools/cbjhfeooiomphlikkblgdageenemhpgc

**Reply by snorkellingcactus** [reply]
I'l take it into account. Althrough i prefer not to deal with a plenty of tools, browser extensions and etc. (just linux, sublime and chrome)

For now i just added cdn tailwind on dev environment (making it redundant since i also need the jit generated css). I'll use only the jit generated one when on staging and prod.
